### PR TITLE
Enable DMTimer PWM pins on BBB and PB

### PR DESCRIPTION
**Pull request by daschachter**
**Scope of Change**
As referenced in #229, there are additional PWM pins available for use on the BeagleBoard. These pins take advantage of the DMTimer to create PWM signals. The current version of `Adafruit_BBIO.PWM` does not have support for these pins. This PR adds support for these pins without change the structure of `Adafruit_BBIO.PWM` from the user's perspective (see the "Tests/Examples" section of this PR).

To add support for these pins, a different PWM path is used when accessing the DMTimer pins. To elaborate, when a DMTimer pin is accessed, its properties can be changed in `/sys/devices/platform/dmtimer-pwm-*/pwm/pwmchip*/pwm0/` (after exporting the pin within the pwmchip directory). This is different from the ocp path used for the standard PWM pins.

Fortunately, because of the structure of this library -- especially the `pwm_exp` struct (which stores the PWM config files) -- the majority of the changes are localized within the `pwm_setup` function of `c_pwm.c`. Minor modifications were also made to the tables found in `common.c`.

**Limitations**
The current changes are meant to support the PocketBeagle and BeagleBone Black. The code has only been tested on a PocketBeagle (see the "Tests/Examples" section). See #229 for a list of kernels that support PWM output from the DMTimer pins. In order to use these pins, the proper device tree overlays must be loaded ([these](https://github.com/beagleboard/bb.org-overlays/pull/194) for PocketBeagle and [these](https://github.com/beagleboard/bb.org-overlays/commit/563af291b72c97a5f93d920ff599e99e74b0c88d) for BeagleBone Black)

**Tests/Examples**
This feature has only been tested on a PocketBeagle (due to the availability of resources), but should also work on the BeagleBone Black. To test the functionality, I loaded `PB-PWM-TIMER-P1.20.dtbo,` connected an LED to P1_20, and ran the following in my Python shell:

    import Adafruit_BBIO.PWM as PWM  
    PWM.start("P1_20", 50) # LED turns on
    PWM.set_duty_cycle(10) # LED dims
    PWM.stop("P1_20") # LED turns off

I repeated this for the other four DMTimer pins on the PocketBeagle.

This example illustrates how the library itself has not changed from the user's perspective (this is the same code that would be used for any PWM pin). I've run this test on the currently recommended Buster IoT image ("AM3358 Debian 10.3 2020-04-06 4GB SD IoT"), as well as the 2020-07-13, Buster IoT testing image "bone-debian-10.4-iot-armhf-2020-07-13-4gb.img.xz".

### CODE REVIEW

**Comment by daschachter** [comment]
@flosincapite

File: source/c_pwm.c [hunk 1]
```diff
@@ -94,6 +94,17 @@ void export_pwm(struct pwm_exp *new_pwm)
     }
 }
 
+int is_dmtimer_pin(pwm_t *p) {
+    if(strlen(p->module) < 5)
```
Comment (flosincapite) [inline source/c_pwm.c:98]: Do we need the length check? I think `strcmp` will do something sane even if `p->module` is short, no?

Comment (daschachter) [reply source/c_pwm.c]: That's a good point -- I originally included the length check for `strncpy`, but it looks like `strncpy`'s behavior will still be [well-defined](http://www.cplusplus.com/reference/cstring/strncpy/) without this check. I've removed this conditional in af9ea51.

**Comment by pdp7** [comment]
@daschachter thanks for the PR.  It looks nicely prepared.  I should be able to review and test out today.

**Comment by pdp7** [comment]
ugh!  sorry my browser is misbehaving.

**Pull request merged**